cloud/gke.go

```diff
@@ -0,0 +1,1038 @@
+		banzaiUtils.LogFatalf(banzaiConstants.TagCreateCluster, "GOOGLE_APPLICATION_CREDENTIALS env var is not specified: %s", err)
+		SetResponseBodyJson(c, http.StatusInternalServerError, gin.H{
+			JsonKeyStatus:  http.StatusInternalServerError,
+			JsonKeyMessage: "GOOGLE_APPLICATION_CREDENTIALS env var is not specified",
+		})
+		return false, nil
+	}
+
+	svc, err := GetGoogleServiceClient()
+	if err != nil {
+		SetResponseBodyJson(c, http.StatusInternalServerError, gin.H{
+			JsonKeyStatus:  http.StatusInternalServerError,
+			JsonKeyMessage: err,
+		})
+		return false, nil
+	}
+
+	cc := GKECluster{
+		ProjectID:         request.Properties.CreateClusterGoogle.Project,
+		Zone:              request.Location,
+		Name:              request.Name,
+		NodeCount:         int64(request.Properties.CreateClusterGoogle.Node.Count),
+		CredentialPath:    credentialPath,
+		CredentialContent: string(data),
+		MasterVersion:     request.Properties.CreateClusterGoogle.Master.Version,
+		NodeVersion:       request.Properties.CreateClusterGoogle.Node.Version,
+	}
+
+	banzaiUtils.LogInfof(banzaiConstants.TagCreateCluster, "Cluster request: %v", generateClusterCreateRequest(cc))
+	createCall, err := svc.Projects.Zones.Clusters.Create(cc.ProjectID, cc.Zone, generateClusterCreateRequest(cc)).Context(context.Background()).Do()
+
+	banzaiUtils.LogInfof(banzaiConstants.TagCreateCluster, "Cluster request submitted: %v", generateClusterCreateRequest(cc))
+
+	if err != nil && !strings.Contains(err.Error(), "alreadyExists") {
+		banzaiUtils.LogInfof(banzaiConstants.TagCreateCluster, "Contains error: %s", err)
+		be := getBanzaiErrorFromError(err)
+		SetResponseBodyJson(c, be.StatusCode, gin.H{
+			JsonKeyStatus:  be.StatusCode,
+			JsonKeyMessage: be.Message,
+		})
+		return false, nil
+	} else {
+		banzaiUtils.LogInfof(banzaiConstants.TagCreateCluster, "Cluster %s create is called for project %s and zone %s. Status Code %v", cc.Name, cc.ProjectID, cc.Zone, createCall.HTTPStatusCode)
+	}
+
+	gkeCluster, err := waitForCluster(svc, cc)
+	if err != nil {
+		banzaiUtils.LogErrorf(banzaiConstants.TagCreateCluster, "Cluster create failed", err)
+		be := getBanzaiErrorFromError(err)
+		SetResponseBodyJson(c, be.StatusCode, gin.H{
+			JsonKeyStatus:  be.StatusCode,
+			JsonKeyMessage: be.Message,
+		})
+		return false, nil
+	}
+
+	cluster2Db := banzaiSimpleTypes.ClusterSimple{
+		Name:             request.Name,
+		Location:         request.Location,
+		NodeInstanceType: request.NodeInstanceType,
+		Cloud:            request.Cloud,
+		Google: banzaiSimpleTypes.GoogleClusterSimple{
+			Project:       request.Properties.CreateClusterGoogle.Project,
+			NodeCount:     request.Properties.CreateClusterGoogle.Node.Count,
+			MasterVersion: request.Properties.CreateClusterGoogle.Master.Version,
+			NodeVersion:   request.Properties.CreateClusterGoogle.Node.Version,
+		},
+	}
+
+	if err := database.Save(&cluster2Db).Error; err != nil {
+		DbSaveFailed(c, err, cluster2Db.Name)
```

Comment:
Maybe it is nothing, but here we are swallowing an error

Reply:
The DbSaveFailed function will logging and setting the response body with the error:
https://github.com/banzaicloud/pipeline/blob/master/cloud/base.go#L74